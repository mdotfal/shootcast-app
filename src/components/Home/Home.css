.main-home {
  display: flex;
  flex-direction: space-evenly;
  justify-content: center;
  min-width: 500px;
  overflow: auto;
}