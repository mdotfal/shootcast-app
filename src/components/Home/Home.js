import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Nav from '../Nav/Nav';
import Sidebar from '../Sidebar/Sidebar';
import AppContext from '../../AppContext';
import './Home.css';

class Home extends Component {


  static contextType = AppContext;

  render() {
    let listId = parseInt( this.props.match.params.listId )
    const cities = !listId 
      ? this.context.cities 
      : this.context.cities.filter( item => item.list_id === listId )
    
    return (
      <div className="main">
        <Nav />
        <div className="main-home">
          <div className="sidebar-home">
            <Sidebar 
              cities={ cities }
            />
          </div>
        </div>
      </div>
    );
  }
};

Home.propTypes = {
  cities: PropTypes.array
}

export default Home;