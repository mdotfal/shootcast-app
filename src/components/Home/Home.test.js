import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Home from './Home';
import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';

describe( `Home component`, () => {

  const div = document.createElement( 'div' );
  const defaultProps = {
    match: { params: { list_id: 1 }}
  };
  const cities = [
    { name: "San Francisco", list_id: 1 },
    { name: "Castro Valley", list_id: 2 }
  ];

  it( `renders the UI as expected`, () => {
    const tree = renderer
      .create( 
        <BrowserRouter>
          <Home/>
        </BrowserRouter>)
      .toJSON();
    expect( tree ).toMatchSnapshot();
  });

  it( 'renders without errors', () => {
    ReactDOM.render( 
      <BrowserRouter>
        <Home { ...defaultProps } { ...cities } />
      </BrowserRouter>
      , div );
    ReactDOM.unmountComponentAtNode( div );
  });
});