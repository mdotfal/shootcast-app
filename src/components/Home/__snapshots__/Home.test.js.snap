// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders the UI as expected 1`] = `
<div
  className="main"
>
  <div
    className="nav"
  >
    <div
      className="nav-logo"
    >
      ShootCast
    </div>
    <div
      className="nav-links"
    >
      <a
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        Sign Out
      </a>
    </div>
  </div>
  <div
    className="main-home"
  >
    <div
      className="sidebar-home"
    >
      <div
        className="sidebar"
      >
        <h1>
          Welcome!
        </h1>
        <p>
          ShootCast is an app where you can create custom lists of cities to view the current weather and forecast for any given city.
        </p>
        <div
          className="addlist"
        >
          <form
            className="addlist-form"
            onSubmit={[Function]}
          >
            <label
              htmlFor="inputList"
            >
              *
            </label>
            <input
              id="inputList"
              name="inputList"
              placeholder="Southern California"
              required={true}
              type="text"
            />
            <button
              type="submit"
            >
              Add List
            </button>
          </form>
        </div>
        <div
          className="sidebar-lists-map"
        >
          <h2>
            Click to display list.
          </h2>
          <ul
            className="sidebar-ul"
          />
        </div>
        <br />
        <div
          className="list"
        >
          <div
            className="addcity"
          >
            <form
              className="addcity-form"
              onSubmit={[Function]}
            >
              <fieldset>
                <legend>
                  <label
                    htmlFor="search"
                  >
                    <p>
                      Add a City to your List
                    </p>
                  </label>
                </legend>
                <input
                  id="search"
                  name="search"
                  placeholder="City or Zip Code"
                  required={true}
                  type="text"
                />
                <label
                  htmlFor="listId"
                >
                  <p>
                    Select a list.
                  </p>
                </label>
                <select
                  id="listId"
                  name="listId"
                >
                  <option>
                    Choose List
                  </option>
                </select>
                <button
                  type="submit"
                >
                  Add City
                </button>
              </fieldset>
            </form>
          </div>
          <p>
            ⇦ Click a city to view Weather ⇨
          </p>
          <div
            className="list-cities"
          >
            <ul />
          </div>
          <div
            className="list-forecast"
          >
            <div
              className="city-name"
            >
              <div
                className="current-weather"
              >
                <h1>
                   
                   
                </h1>
              </div>
            </div>
            <div
              className="forecast-items"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
